convert2rhel/backup.py

```diff
@@ -189,33 +189,55 @@ def pop(self):
 
         return restorable
 
-    def pop_all(self):
+    def pop_all(self, _honor_partitions=False):
         """
         Restores all RestorableChanges known to the Controller and then returns them.
 
         :returns: List of RestorableChange objects that were known to the Controller.
         :raises IndexError: If there are no RestorableChanges currently known to the Controller.
 
         After running, the Controller object will not know about any RestorableChanges.
-        """
-        # This code ignores partitions.  Only restore_to_partition() honors
-        # them.
-        restorables = [r for r in self._restorables if r is not self.partition]
 
-        if not restorables:
+        .. note:: _honor_partitions is part of a hack for 1.4 to let us split restoring changes into
+            two parts.  During rollback, the part before the partition is restored before the legacy
+            backups.  Then the remainder of the Changes managed by BackupController are restored.
+            This can go away once we merge all of the legacy backups into RestorableChanges managed
+            by BackupController.
+        """
+        # Only raise IndexError if there are no restorables registered.
+        # Partitions are ignored for this check as they aren't really Changes.
+        if not self._restorables or all(r == self.partition for r in self._restorables):
             raise IndexError("No backups to restore")
 
-        # We want to restore in the reverse order the changes were enabled.
-        for restorable in reversed(restorables):
-            restorable.restore()
+        # Restore the Changes in the reverse order the changes were enabled.
+        processed_restorables = []
+        while True:
+            try:
+                restorable = self._restorables.pop()
+            except IndexError:
+                break
+
+            if restorable is self.partition:
+                if _honor_partitions:
+                    # Stop once a partition is reached (this is how
+                    # pop_to_partition() is implemented.
+                    return []
+                else:
+                    # This code ignores partitions.  Only pop_to_partition() honors
+                    # them.
+                    continue
 
-        # Reset the internal storage in case we want to use it again
-        self._restorables = []
+            try:
+                restorable.restore()
+            # Catch SystemExit too because we might still be calling
+            # logger.critical in some places.
+            except (Exception, SystemExit) as e:
+                # Don't let a failure in one restore influence the others
+                loggerinst.warning("Error while rolling back a change: %s" % str(e))
```

Comment:
Does it make sense to log what restorable caused the error?

Reply:
It does.  When I wrote this I was thinking we don't have something to do that (like `Action.id`) but I just realized we can get the class name from the RestorableChaneg instance.  Let me look into adding that.

Reply:
Added `restorable.__class__.__name__` to the error message here: https://github.com/oamg/convert2rhel/pull/912/files#diff-8f4622020cde9a253bc82f69bf2b88910bcc5a84943db3c7d3a435becedfd4d5R238